Attendance Class Detail Editing old

Scott123180 opened the pull request:
- added new page for editing class information
- removed confirmation page for editing participants
- changed page permissions and routes for these pages

Fixes #152

Scott123180 commented:
@evanmena 
- requested changes made
- also fixed #166

vmjoseph commented on routes.php:
```diff
@@ -64,4 +64,4 @@
 $router->add('/historical-class-search-results', 'attendance/historical_class_lookup_results.php', 'Historical Class Search Results');
 $router->add('/attendance-form-confirmed', 'attendance/attendance_form_confirmed.php', 'Attendance Confirmed');
 $router->add('/edit-participant', 'attendance/edit_participant.php', 'Edit Participant');
```
This will cause a conflict when the participant management pull request is accepted. Both of the routes will be the same

Scott123180 replied:
@vmjoseph 

I see where the conflict is, but this route has been there since attendance was merged into the branch. I think edit-participant may be too vague of a name, so I will change it to `attendance-edit-participant`. However, I think that the fact that we had this conflict means the page name `edit-participant` may not be detailed enough. I would suggest adding more detail to the page names. Perhaps `ps-edit-participant`.

vmjoseph replied:
Understood. I'll make the addition to my end

evanmena commented on views/attendance/new_class.php:
```diff
@@ -103,28 +103,21 @@ function populateTimes() {
-            } else{
-                hourAM = i;
+        //populate am first, then pm
+        for(var z = 0; z <= 1; z++) {
```
This isn't what I meant. `optionArray` appears to be just an array holding values that you later append to `selectTime`. Just append as you go instead of storing them in an array first. `selectTime.appendChild(createTime(hourAM, "00", amORpm))` is plenty readable and you won't need to loop on line 157 to append `optionArray[k]`

evanmena replied:
The loop you did add is an okay addition though and reduces some duplicate code.

Scott123180 commented:
@evanmena 

- implemented requested changes
- changed route for `edit-participant` as per conflict in route names with Vallie
- reduced duplicated code for 'populateTimes()' function by moving to 'attendance-utilities.js' file

blackjack26 commented on views/attendance/attendance_edit_class_info.php:
```diff
@@ -0,0 +1,318 @@
+ * @since [initial version number]
+ */
+
+authorizedPage();
```
Don't need the `authorizedPage()` function since this is handled in the `page_permissions.ini`.

blackjack26 commented on views/attendance/attendance_edit_class_info.php:
```diff
@@ -0,0 +1,318 @@
+include ('attendance_utilities.php');
+
+//form options
+$result_curriculum = $db->no_param_query("SELECT c.curriculumid, c.curriculumname FROM curricula c ORDER BY c.curriculumname ASC;");
```
Make sure to only get current curricula:

```sql
SELECT c.curriculumid, c.curriculumname
FROM curricula c
WHERE c.df IS FALSE
ORDER BY c.curriculumname ASC
```

Scott123180 replied:
need all of them for creating my `classesMatrix` which contains all of the topic names, the id's for those topics, and the curriculum they belong to.

Scott123180 replied:
`IS FALSE` results in a syntax error, but I can do `c.df = 0`

blackjack26 commented on views/attendance/attendance_edit_class_info.php:
```diff
@@ -0,0 +1,318 @@
+//form options
+$result_curriculum = $db->no_param_query("SELECT c.curriculumid, c.curriculumname FROM curricula c ORDER BY c.curriculumname ASC;");
+
+$result_classes = $db->no_param_query("SELECT cc.curriculumid, topicname, cc.classid FROM curriculumclasses cc, classes WHERE classes.classid = cc.classid ORDER BY cc.curriculumid;");
```
Same thing here, check for `df IS FALSE`

blackjack26 commented on views/attendance/attendance_edit_class_info.php:
```diff
@@ -0,0 +1,318 @@
+    "FROM people peop, employees emp, facilitators f " .
+    "WHERE peop.peopleid = emp.employeeid " .
+    "AND emp.employeeid = f.facilitatorid " .
+    "AND f.df = 0 " .
```
With the updated database this should be `f.df IS FALSE`

Scott123180 replied:
@blackjack26 what DB has the update? I'm connected to 10.11.12.34 and is false isn't working

Scott123180 commented:
@blackjack26 
- removed the authorizedPage function from every page that had it
- added df flag to location across two pages, `attendance_edit_class_info` and `new_class`
- changed the df to is false on the two pages that contained it. This df flag can't be tested until 10.11.12.34 gets updated and will break the attendance app until it does

blackjack26 commented:
I can test it on my local which has the up to date stuff

evanmena commented on public/js/attendance-scripts/attendance-utilities.js:
```diff
@@ -0,0 +1,39 @@
+function populateTimes() {
+    var selectTime = document.getElementById('time-input');
+
+    var optionArray = [];
```
Remove this unused array. You're appending directly to the `time-input` element now.

evanmena commented on public/js/attendance-scripts/attendance-utilities.js:
```diff
@@ -0,0 +1,39 @@
+        }
+    }
+
+    for(var k = 0; k < optionArray.length; k++){
```
Dead code, optionArray is not used anymore.

Scott123180 commented:
@evanmena 
My bad, sloppy coding. Fixed now

evanmena commented on public/js/attendance-scripts/attendance-form-add-new-person.js:
```diff
@@ -107,14 +107,14 @@ function validateAge(age) {
     if(typeof(ageNumber) !== "number") return false;
     //returns true if age in valid range 18-100
-    return((ageNumber >= 18) && (ageNumber <= 100));
+    return((ageNumber >= 1) && (ageNumber <= 130));
```
Just check if `ageNumber > 0` there's no need to make up some upper value, we don't have a requirement. Same for `numChildren >= 0`. It may seem pedantic but the minimum needed here is just a positive number.

evanmena commented on public/js/attendance-scripts/attendance-form-add-new-person.js:
```diff
@@ -107,14 +107,14 @@ function validateAge(age) {
     if(isNaN(ageNumber)) return false;
     if(typeof(ageNumber) !== "number") return false;
     //returns true if age in valid range 18-100
```
Remove or update this comment

evanmena commented on public/js/attendance-scripts/attendance-utilities.js:
```diff
@@ -0,0 +1,33 @@
+    for(var z = 0; z <= 1; z++) {
+        var amORpm = (z === 0) ? "AM" : "PM";
+
+        //AM
```
Remove `//AM` and rename `var hourAM` to just `hour` since you're covering both cases in this loop now.

Scott123180 commented:
@evanmena just did a git amend
